packages/@ember/debug/lib/warn.ts

```diff
@@ -106,7 +111,7 @@ if (DEBUG) {
     assert(missingOptionsDeprecation, Boolean(options));
     assert(missingOptionsIdDeprecation, Boolean(options && options.id));
 
-    invoke('warn', message, test, options);
+    invoke('warn', message, test as boolean, options);
```

Comment:
Let's add a `SAFETY` comment here explaining why the cast is safe.

```suggestion
    // SAFETY: we have explicitly assigned `false` if the user invoked the
    // arity-2 version of the overload, so we know `test` is always either
    // `undefined` or a `boolean` for type-safe callers.
    invoke('warn', message, test as boolean | undefined, options);
```